pkg/registry/common/refresh/nse_registry_client_test.go

```diff
@@ -18,171 +18,187 @@ package refresh_test
 	"google.golang.org/grpc"
 	"google.golang.org/protobuf/types/known/timestamppb"
 
+	"github.com/networkservicemesh/api/pkg/api/registry"
+
 	"github.com/networkservicemesh/sdk/pkg/registry/common/refresh"
+	"github.com/networkservicemesh/sdk/pkg/registry/core/adapters"
+	"github.com/networkservicemesh/sdk/pkg/registry/core/next"
 )
 
 const testExpiryDuration = time.Millisecond * 100
 
-type testNSEClient struct {
-	sync.Mutex
-	requestCount int
+func testNSE(name string) *registry.NetworkServiceEndpoint {
+	return &registry.NetworkServiceEndpoint{
+		Name: name,
+	}
 }
 
-func (t *testNSEClient) Register(ctx context.Context, in *registry.NetworkServiceEndpoint, opts ...grpc.CallOption) (*registry.NetworkServiceEndpoint, error) {
-	t.Lock()
-	defer t.Unlock()
-	t.requestCount++
-	return next.NetworkServiceEndpointRegistryClient(ctx).Register(ctx, in, opts...)
-}
+func TestNewNetworkServiceEndpointRegistryClient(t *testing.T) {
+	defer goleak.VerifyNone(t, goleak.IgnoreCurrent())
 
-func (t *testNSEClient) Find(ctx context.Context, in *registry.NetworkServiceEndpointQuery, opts ...grpc.CallOption) (registry.NetworkServiceEndpointRegistry_FindClient, error) {
-	panic("implement me")
-}
+	countClient := new(requestCountClient)
+	client := next.NewNetworkServiceEndpointRegistryClient(
+		refresh.NewNetworkServiceEndpointRegistryClient(refresh.WithDefaultExpiryDuration(testExpiryDuration)),
+		countClient,
+	)
 
-func (t *testNSEClient) Unregister(ctx context.Context, in *registry.NetworkServiceEndpoint, opts ...grpc.CallOption) (*empty.Empty, error) {
-	return nil, nil
-}
+	_, err := client.Register(context.Background(), &registry.NetworkServiceEndpoint{
+		Name: "nse-1",
+	})
+	require.NoError(t, err)
 
-type checkExpirationTimeClient struct{ *testing.T }
+	require.Eventually(t, func() bool {
+		return atomic.LoadInt32(&countClient.requestCount) > 1
+	}, time.Second, testExpiryDuration/4)
 
-func (c *checkExpirationTimeClient) Register(ctx context.Context, in *registry.NetworkServiceEndpoint, opts ...grpc.CallOption) (*registry.NetworkServiceEndpoint, error) {
-	require.NotNil(c, in.ExpirationTime)
-	return in, nil
+	_, err = client.Unregister(context.Background(), testNSE("nse-1"))
+	require.NoError(t, err)
 }
 
-func (c *checkExpirationTimeClient) Find(ctx context.Context, in *registry.NetworkServiceEndpointQuery, opts ...grpc.CallOption) (registry.NetworkServiceEndpointRegistry_FindClient, error) {
-	panic("implement me")
-}
+func TestRefreshNSEClient_ShouldSetExpirationTime_BeforeCallNext(t *testing.T) {
+	defer goleak.VerifyNone(t, goleak.IgnoreCurrent())
 
-func (c *checkExpirationTimeClient) Unregister(ctx context.Context, in *registry.NetworkServiceEndpoint, opts ...grpc.CallOption) (*empty.Empty, error) {
-	return new(empty.Empty), nil
-}
+	client := next.NewNetworkServiceEndpointRegistryClient(
+		new(requestCountClient),
+		refresh.NewNetworkServiceEndpointRegistryClient(refresh.WithDefaultExpiryDuration(time.Hour)),
+		&checkNSEClient{
```

Comment:
I think this should be moved to the same place as it is in `netowrkservice/`

Reply:
Yep, moved.